src/test/java/hudson/plugins/deploy/PipelineSyntaxTest.java

```diff
@@ -0,0 +1,105 @@
+package hudson.plugins.deploy;
+
+import hudson.model.Label;
+import hudson.model.Result;
+import hudson.model.labels.LabelOperatorPrecedence;
+import hudson.model.labels.LabelVisitor;
+import hudson.util.VariableResolver;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * Tests pipeline compatibility. Since there are no *.war files in the workspace all of these scripts
+ * will ultimately result in a no-op.
+ */
+public class PipelineSyntaxTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private String getFullScript (String func) {
+        return "node {\n" +
+                    "writeFile(file: 'readme.txt', text: 'this creates a workspace if one doesnt already exist')\n" +
+                    func +
+                "}";
+    }
+
+    @Test
+    public void testNoAdapterDeploy() throws Exception {
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "DryRunTest");
+        j.createOnlineSlave(Label.get("remote"));
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript("deploy(war: 'target/app.war', contextPath: 'app', onFailure: false)"),
+                false));
+        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+    }
+
+    @Test
+    public void testMockAdapterDeploy() throws Exception {
+        j.jenkins.setNumExecutors(1);
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "MockTest");
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript("deploy(adapters: [mock()], war: 'target/app.war', contextPath: 'app')"),
```

Comment:
Here is a problem. You invoke a Pipeline-incompatible adapter from Pipeline and expect the build to pass... Sound like a test for a bad behavior.

I would rather recommend adding proper error propagation for this case. E.g. the default adapter method check if it is called for non-AbstractProject type and then throw error